tests/UnifiedSpecTests/Context.php

```diff
@@ -316,6 +325,79 @@ private function createClient(string $id, stdClass $o): void
         $this->entityMap->set($id, FunctionalTestCase::createTestClient($uri, $uriOptions, $driverOptions));
     }
 
+    private function createClientEncryption(string $id, stdClass $o): void
+    {
+        Util::assertHasOnlyKeys($o, [
+            'id',
+            'clientEncryptionOpts',
+        ]);
+
+        $clientEncryptionOpts = [];
+        $clientId = null;
+
+        if (isset($o->clientEncryptionOpts)) {
+            assertIsObject($o->clientEncryptionOpts);
+            $clientEncryptionOpts = (array) $o->clientEncryptionOpts;
+        }
+
+        if (isset($clientEncryptionOpts['keyVaultClient'])) {
+            assertIsString($clientEncryptionOpts['keyVaultClient']);
+            /* Record the keyVaultClient's ID, which we'll later use to track
+             * the parent client in the entity map. */
+            $clientId = $clientEncryptionOpts['keyVaultClient'];
+            $clientEncryptionOpts['keyVaultClient'] = $this->entityMap->getClient($clientId)->getManager();
+        }
+
+        if (isset($clientEncryptionOpts['kmsProviders'])) {
```

Comment:
Noting for the benefit of @levon80999 and others that may want to run these tests: I suggest defining these secrets in a `phpunit.xml` file based on `phpunit.xml.dist`. This file takes precedence over the dist file and is in gitignore, allowing your own configuration to be added.